tools/ci_build/github/azure-pipelines/templates/jobs/download_linux_qnn_sdk.yml

```diff
@@ -21,16 +21,18 @@ steps:
       sdk_file="$(QnnSDKRootDir)/sdk.yaml"
       qnn_sdk_version=$(grep '^version:' "$sdk_file" | head -n 1 | cut -d':' -f2 | xargs | cut -d'.' -f1-3)
       if [[ -n "$qnn_sdk_version" ]]; then
-        echo "##vso[task.setvariable variable=QnnSDKVersionYaml;isOutput=true]$qnn_sdk_version"
+        # Do not output ##vso[] commands with `set -x` or they may be parsed again and include a trailing quote.
```

Comment:
Should we do a sanity check that parameters.QnnSDKVersion matches? that _should_ always be the case, but if for some strange reason it was not it would be pretty hard to debug if the CI is using QnnSDKVersion everywhere but the actual SDK version differs.

Reply:
Makes sense, I will add a sanity check in `download_linux_qnn_sdk.yml` but use `parameters.QnnSDKVersion` in the test stage. So no new variable will be set in `download_linux_qnn_sdk.yml`